pkg/sql/delegate/show_table.go

```diff
@@ -52,6 +54,14 @@ WHERE name = %s
 	return parse(fmt.Sprintf(showCreateQuery, lexbase.EscapeSQLString(n.Name.Object())))
 }
 
+func (d *delegator) delegateShowCreateSecondaryIndexes(n *tree.ShowCreate) (tree.Statement, error) {
+	sqltelemetry.IncrementShowCounter(sqltelemetry.Indexes)
+
+	const showCreateSecondaryIndexFromTable = `SELECT create_statement FROM crdb_internal.table_indexes WHERE descriptor_id = %[2]s::regclass::int AND index_type != 'primary'`
```

Comment:
i think this doesn't work when you use `SHOW CREATE SECONDARY INDEXES FROM <other_db>.table`. can you try it on a table on another database in the lgoic tests?

to fix this, you'll need to follow how `SHOW CREATE TABLE` references `<database_name>.crdb_internal.table_indexes` which will fetch the correct result